packages/protocol/contracts/common/Accounts.sol

```diff
@@ -0,0 +1,483 @@
+  )
+    external
+  {
+    if(!isAccount(msg.sender)) {
+      createAccount();
+    }
+    setName(name);
+    setAccountDataEncryptionKey(dataEncryptionKey);
+    setWalletAddress(walletAddress);
+  }
+
+  /**
+   * @notice Authorizes an address to sign votes on behalf of the account.
+   * @param voter The address of the vote signing key to authorize.
+   * @param v The recovery id of the incoming ECDSA signature.
+   * @param r Output value r of the ECDSA signature.
+   * @param s Output value s of the ECDSA signature.
+   * @dev v, r, s constitute `voter`'s signature on `msg.sender`.
+   */
+  function authorizeVoteSigner(
+    address voter,
+    uint8 v,
+    bytes32 r,
+    bytes32 s
+  )
+    external
+    nonReentrant
+  {
+    Account storage account = accounts[msg.sender];
+    authorize(voter, account.signers.voting, v, r, s);
+    account.signers.voting = voter;
+    emit VoteSignerAuthorized(msg.sender, voter);
+  }
+
+  /**
+   * @notice Authorizes an address to sign consensus messages on behalf of the account.
+   * @param validator The address of the signing key to authorize.
+   * @param v The recovery id of the incoming ECDSA signature.
+   * @param r Output value r of the ECDSA signature.
+   * @param s Output value s of the ECDSA signature.
+   * @dev v, r, s constitute `validator`'s signature on `msg.sender`.
+   */
+  function authorizeValidationSigner(
+    address validator,
+    uint8 v,
+    bytes32 r,
+    bytes32 s
+  )
+    external
+    nonReentrant
+  {
+    Account storage account = accounts[msg.sender];
+    authorize(validator, account.signers.validating, v, r, s);
+    account.signers.validating = validator;
+    emit ValidationSignerAuthorized(msg.sender, validator);
+  }
+
+  /**
+   * @notice Authorizes an address to sign attestations on behalf of the account.
+   * @param attestor The address of the signing key to authorize.
+   * @param v The recovery id of the incoming ECDSA signature.
+   * @param r Output value r of the ECDSA signature.
+   * @param s Output value s of the ECDSA signature.
+   * @dev v, r, s constitute `attestor`'s signature on `msg.sender`.
+   */
+  function authorizeAttestationSigner(
+    address attestor,
+    uint8 v,
+    bytes32 r,
+    bytes32 s
+  )
+    public
+  {
+    Account storage account = accounts[msg.sender];
+    authorize(attestor, account.signers.attesting, v, r, s);
+    account.signers.attesting = attestor;
+    emit AttestationSignerAuthorized(msg.sender, attestor);
+  }
+
+  /**
+   * @notice Returns the account associated with `accountOrAttestationSigner`.
+   * @param accountOrAttestationSigner The address of the account or active authorized attestation
+                                       signer.
+   * @dev Fails if the `accountOrAttestationSigner` is not an account or active authorized
+          attestation signer.
+   * @return The associated account.
+   */
+  function getAccountFromActiveAttestationSigner(address accountOrAttestationSigner)
```

Comment:
Please put view functions at the end of the contract. When we upgraded our linter this check was disabled but I'd like to turn it back on